app/controllers/podcast_metrics_controller.rb

```diff
@@ -96,17 +197,77 @@ def metrics_params
       )
   end
 
+  def uniques_params
+    params
+      .permit(:uniques_selection)
+      .with_defaults(
+        uniques_selection: "last_7_rolling"
+      )
+      .merge(metrics_params)
+  end
+
+  def dropdays_params
+    params
+      .permit(:dropday_range)
+      .with_defaults(
+        dropday_range: 7
+      )
+      .merge(metrics_params)
+  end
+
   def episode_rollups(episodes, rollups, totals)
-    episodes.map do |ep|
+    episodes.to_enum(:each_with_index).map do |episode, i|
       {
-        ep: ep,
+        episode: episode,
         rollups: rollups.select do |r|
-          r["episode_id"] == ep.guid
+          r["episode_id"] == episode.guid
         end,
         totals: totals.select do |r|
-          r["episode_id"] == ep.guid
-        end
+          r["episode_id"] == episode.guid
+        end,
+        color: colors[i]
       }
     end
   end
+
+  def episode_dropdays(episodes, dropdays)
+    episodes.to_enum(:each_with_index).map do |episode, i|
+      {
+        episode: episode,
+        dropdays: dropdays.select do |d|
+          d["episode_id"] == episode.guid
+        end,
+        color: colors[i]
+      }
+    end
+  end
+
+  def generate_date_range(date_start, date_end, interval)
+    start_range = date_start.to_datetime.utc.send(:"beginning_of_#{interval.downcase}")
+    end_range = date_end.to_datetime.utc.send(:"beginning_of_#{interval.downcase}")
+    range = []
+    i = 0
+
+    while start_range + i.send(:"#{interval.downcase.pluralize}") <= end_range
+      range << start_range + i.send(:"#{interval.downcase.pluralize}")
+      i += 1
+    end
+
+    range
+  end
+
+  def colors
```

Comment:
colors were moved from the helper to here. i wanted to make sure they were just in one place but each of the different charts were pulling from the same set.